netplan_cli/cli/utils.py

```diff
@@ -196,10 +196,43 @@ def get_interface_driver_name(interface, only_down=False):  # pragma: nocover (c
     return driver_name
 
 
-def get_interface_macaddress(interface):
-    # return an empty list (and string) if no LL data can be found
-    link = netifaces.ifaddresses(interface).get(netifaces.AF_LINK, [{}])[0]
-    return link.get('addr', '')
+def _get_permanent_macaddress(interface: str) -> str:
+    mac = ""
+    try:
+        out = subprocess.check_output(['ethtool', '-P', interface]).decode('utf-8')
+        split = out.split(': ')
+        if len(split) == 2 and split[1].strip() != 'not set':
```

Comment:
nitpick: are we sure the "not set" string might not change in the future, or introduce other options?

Maybe it would be safer to actually use `cli.utils.is_valid_macaddress` to validate the output.